nucleus/modelci/utils.py

```diff
@@ -0,0 +1,42 @@
+from .unit_test_evaluation import (
+    UnitTestEvaluation,
+    UnitTestItemEvaluation,
+    UnitTestEvaluationStatus,
+)
+
+
+def try_convert_float(float_str: str) -> float:
```

Comment:
`NaN` has a [special meaning in the definition of floats](https://en.wikipedia.org/wiki/NaN), e.g. when you do `0 / 0`. I think we want to surface some other kind of error string if someone passes in something that wasn't even a valid float to begin with.

Reply:
I do think we should still fail gracefully, but maybe I can just log an error as well?

Reply:
Not sure if we need to log stuff in user code, that should be on them (it does bring up the question of what kind of API we want to provide for user-defined logging, if any). Though if we're provided any pre-built code, then I think it can make sense to log ourselves.

We should still be able to fail gracefully while still being type-correct. I think this stems from `UnitTestItemEvaluation` having `result` being a `float`, when IMO it should be an `Optional[float]`. The db table should correspondingly have the column be `nullable`, to signify an exception in the code (maybe you'd need an extra `nullable` column for an error code/string). Overloading `NaN` to mean both an exception and `0 / 0` could be a confusing edge case.

Reply:
Ah ok `Optional[float]` makes sense - sounds good.